feat: use app wildcards for apps if configured

deansheather opened the pull request:
- Adds `subdomain` to the WorkspaceApp struct.
- Changes the dashboard to use subdomain app access if `subdomain` is `true` and an app hostname is set.
- Add a tooltip explaining why the app button is greyed out if so.
- Upgrade coder/coder terraform module to 0.5.0 (which deprecates `relative_path` and adds `subdomain`).
- Rename `relative_path` to `subdomain` across the codebase and in the database

deansheather committed:
feat: use app wildcards for apps if configured

deansheather committed:
fixup! feat: use app wildcards for apps if configured

deansheather committed:
feat: working tooltips on disabled app buttons

deansheather committed:
feat: default apps relative_path to true

jsjoeio commented on provisioner/terraform/testdata/multiple-apps/multiple-apps.tf:
```diff
@@ -12,19 +12,30 @@ resource "coder_agent" "dev1" {
   arch = "amd64"
 }
 
+# app1 is for testing relative_path default.
```
👏🏼 Great comments!

jsjoeio reviewed (approved):
Nice work!

kylecarbs commented on codersdk/workspaceapps.go:
```diff
@@ -21,6 +21,11 @@ type WorkspaceApp struct {
+	// `coder server` or via a hostname-based dev URL. If this is set to false
+	// and there is no app wildcard configured on the server, the app will not
+	// be accessible in the UI.
+	RelativePath bool `json:"relative_path"`
```
I think we'll have to invert this on the provider to not break current deployments.

This is defaulting to `false`: https://github.com/coder/terraform-provider-coder/blob/main/provider/app.go#L62, which I believe would break everyone.

eg. remove `relative_path` and add `subdomain` instead

deansheather replied:
done!

deansheather committed:
feat: relative_path -> subdomain

- rename relative_path -> subdomain when referring to apps
    - migrate workspace_apps.relative_path to workspace_apps.subdomain
- upgrade coder/coder terraform module to 0.5.0

deansheather committed:
fixup! feat: relative_path -> subdomain

deansheather committed:
fixup! feat: relative_path -> subdomain

deansheather committed:
fixup! feat: relative_path -> subdomain

kylecarbs commented on provisioner/terraform/resources.go:
```diff
@@ -27,12 +25,14 @@ type agentAttributes struct {
+	Command   string `mapstructure:"command"`
+	Subdomain bool   `mapstructure:"subdomain"`
+	// RelativePath is deprecated in favor of Subdomain.
+	RelativePath *bool                      `mapstructure:"relative_path"`
```
I don't think this needs to be a pointer anymore

deansheather replied:
We prefer the deprecated value over the new value if it's set, so we have to use a pointer for this to determine if it was explicitly set or not. I will add a comment

deansheather committed:
fixup! feat: relative_path -> subdomain

deansheather committed:
Merge branch 'main' into dean/app-relative-path

deansheather committed:
empty commit to make github actions work
